File: programmers.co.kr/stack-queue/42586/main.go
package main

import (
	"fmt"
)

func main() {
	var progresses = []int{93, 30, 55}
	var speeds = []int{1, 30, 5}
	var day []int
	var result []int

	for i := 0; i < len(progresses); i++ {
		day = append(day, (100-progresses[i])/speeds[i])
		if (100-float32(progresses[i]))/float32(speeds[i]) > float32((100-progresses[i])/speeds[i]) {
			day[i]++
		}
	}

	temp := 0

	for i, v := range day {
		if i == 0 {
			if v < 1 {
				break
			}

			result = append(result, 1)
			temp = v
			continue
		}

		if v <= temp {
			result[len(result)-1]++
		} else {
			result = append(result, 1)
			temp = v
		}
	}

	fmt.Println(result)
}

File: programmers.co.kr/stack-queue/42586/main.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
	vector<int> answer;
	vector<int> days;

	int size = progresses.size();

	for (int i = 0; i < size; i++) {
		int day = (100 - progresses[i]) / speeds[i];

		if ((100 - progresses[i]) % speeds[i] > 0) {
			day++;
		}

		days.push_back(day);
	}

	int count = 0;
	int temp = 0;
	size = days.size();
	for (int i = 0; i < size; i++) {
		if (i != 0 && temp >= days[i]) {
			answer[answer.size() - 1]++;
		}
		else {
			answer.push_back(1);
			temp = days[i];
		}
	}
	return answer;
}